lib/markdown2.py

```diff
@@ -391,8 +391,20 @@ def convert(self, text):
         text += "\n"
 
         rv = UnicodeWithAttrs(text)
-        if "toc" in self.extras:
+        if ("toc" in self.extras or "inline-toc" in self.extras):
+            # Generate TOC HTML as a property to be able to hijack it in "inline-toc" for subsctitution
+            # TODO (Tomas Hubelbauer): See about using that as a static method without a need for a throwaway instance.
```

Comment:
This is currently needlessly inefficient, here I just take advantage of the fact that the object with `toc_html` on it is already created and replace its reference with a new copy. I will investigate static methods in Python to see if I can run `inline-toc` before [line 393](https://github.com/trentm/python-markdown2/pull/280/files#diff-8562cc594a57b4deeda3ed2cb6208ba7R393) and avoid have to create two objects.

Reply:
I did not end up using `staticmethod` decorator because I wasn't sure if Python 2.2 and 2.3 are expected to be supported and even then it gave me a hard time with me calling it within `UnicodeWithAttrs` where it was defined but it reporting `UnicodeWithAttrs.toc_html` not existing.